numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -1792,8 +1792,12 @@ type_tuple_userloop_type_resolver(PyUFuncObject *self,
             if (obj == NULL) {
                 continue;
             }
-            funcdata = (PyUFunc_Loop1d *)NpyCapsule_AsVoidPtr(obj);
-            while (funcdata != NULL) {
+
+            for (
+                funcdata = (PyUFunc_Loop1d *)NpyCapsule_AsVoidPtr(obj);
+                funcdata != NULL;
+                funcdata = funcdata->next
+            ) {
```

Comment:
I'd much prefer the initialization to remain in the `for`, but I'm happy to change the wrapping to match the above